### PR TITLE
Make use session sync, use a plugin based approach for session fetching

### LINKED ISSUE #65: Proposal to improve the experience of `useSession`
### Describe the feature

Hi! I'm João Pedro. I'm one of the maintainers of the nuxt 2 auth module :)

So, currently, each time `useSession` is used it makes a request to the api to get the session, which means we have to `await` for the request and this makes the component asynchronous. And if it's used in more than one component, it will make unnecessary requests.

My proposal is to use `useSession` only to get the data stored with `useState`, and fetch the session on load the page through a plugin.

This follows the same approach of the auth module, which fetches the user on load the auth plugin.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by BracketJohn**
Closes #65

This PR:
- adds a plugin to maintain the session life cycle
- adds a global auth-middleware
- makes `useSession` sync
- adds `lastRefreshedAt` as a public property 
- cleans up the structure of the module a bit:
  - add utils folder,
  - factor all `useState` into a single place to make it easier to maintain + update

Still to do:
- [x] document plugin usage / puropose
- [x] showcase full useSession properties in docs
- [x] update configuration section
- [x] adapt useSession usage everywhere
- [x] make guest mode clearer / more documented -> removed it
- [x] rename refetchInterval

for later work to update https://github.com/sidebase/nuxt-auth-example:
- [ ] bump to new version once released
- [ ] remove inline middleware
- [ ] be aware that `getCsrfToken` changed

for new issues / out of scope:
- [ ] do not register `useSessionState` composable
- [ ] document how to expand user object
- [ ] add a global server-middleware
- add broadcasting to sync session between tabs (opened #70)

Checklist:
- [x] issue number linked above after pound (`#`)
    - replace "Closes " with "Contributes to" or other if this PR does not close the issue
- [x] manually checked my feature / checking not applicable
- [x] wrote tests / testing not applicable
- [x] attached screenshots / screenshot not applicable

### CODE REVIEW

File: src/runtime/composables/useSession.ts [hunk 1]
```diff
@@ -317,12 +253,10 @@ export default async (initialGetSessionOptions: UseSessionOptions = {}) => {
...
   }
 
-  const getters: {
-    status: Ref<SessionStatus>,
-    data: Ref<SessionData>
-  } = {
-    status,
-    data
+  const getters = {
+    status: readonly(status),
```
Comment (JoaoPedroAS51) [inline src/runtime/composables/useSession.ts:257]: As `status` is now `computed`, we don't need to add `readonly`

**Review by JoaoPedroAS51 — COMMENTED** [review]
Awesome!!! Nice work!! 😃 

I just pointed out a thing here and there.

File: src/runtime/utils/url.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { joinURL, parseURL } from 'ufo'
+import _getURL from 'requrl'
+import { useRequestEvent } from '#app'
+import { useRuntimeConfig, navigateTo as _navigateTo } from '#imports'
+
+const _getBasePath = () => parseURL(useRuntimeConfig().public.auth.url).pathname
```
[the remark this answers is not in the document]
Comment (BracketJohn) [reply src/runtime/utils/url.ts:6]: Originally, I did this to ensure that when the `origin` is set, it's strictly used on the app- and server-side. However, that doesn't make any sense as on the app-side anyone can edit the js/html/... anyways. 

I'll use the `useRequestEvent` here but keep using the origin in the auth handler.

[the remark this answers is not in the document]
Comment (JoaoPedroAS51) [reply src/runtime/utils/url.ts:6]: Maybe we can create a func that uses the origin defined by the user and if not defined fallback to `requrl`. Makes sense?

[the remark this answers is not in the document]
Comment (BracketJohn) [reply src/runtime/utils/url.ts]: I thought on the server-side we need it for security reasons, but you may be right that we can drop it. Looking through the next auth docs and code I cannot find a place that requires it, only the `secret` / `NEXTAUTH_SECRET`.

Let's investige + remove this in another issue.

[the remark this answers is not in the document]
Comment (BracketJohn) [reply src/runtime/utils/url.ts]: I've opened #72 foer this (:

**Review by JoaoPedroAS51 — APPROVED** [review]

**Pull request merged**